include/revng/ABIAnalyses/UsedReturnValuesOfFunctionCall.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Value.h"
+
+#include "revng/ABIAnalyses/Common.h"
+#include "revng/BasicAnalyses/GeneratedCodeBasicInfo.h"
+#include "revng/MFP/MFP.h"
+#include "revng/StackAnalysis/StackAnalysis.h"
+#include "revng/Support/revng.h"
+
+namespace UsedReturnValuesOfFunctionCall {
+
+enum State { Unknown, Maybe, Yes };
+
+// special register index, if found inside LatticeElement
+// it means that it is the first time visiting the BasicBlock
+const int32_t InitialRegisterState = -1;
```

Comment:
In a monotone framework, nodes not visited yet are initialized to bottom (unless they're extremal points).
Put bottom in your lattice.

Reply:
Yes this is a hack to support `GraphTraits` where we need to know if this is really an unvisited node (i.e. we need a value that will never be produced by any transfer function).

Ex. for `GraphTrats<BasicBlock*>` and analyses that start from a `CallSite` (i.e. potentially in the middle of a `BasicBlock`) and have a special transfer function associated to that in case we reach the call again via a back edge

This value was put to keep using a standard container (`Map` like data structures), but turning it to a flag in a `struct` would work too

Reply:
Resolved requiring `CallSites` to be in their own basic block